requires.py

```diff
@@ -0,0 +1,58 @@
+from charms.reactive import when, when_not
+from charms.reactive import set_flag, clear_flag
+from charms.reactive import Endpoint
+
+class MountRequires(Endpoint):
+
+    @when('endpoint.{endpoint_name}.changed')
+    def changed(self):
+        if any(unit.received['mountpoint'] for unit in self.all_units):
```

Comment:
This technically should be `received_raw`, and `all_units` is deprecated in favor of `all_joined_units`, but neither of those will actually cause any problem.